ts/WoltLabSuite/Core/Helper/ClickGuard.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Prevents concurrent runs of the event handler for the click event by blocking
+ * the event while a previous call is still running.
+ *
+ * @author Alexander Ebert
+ * @copyright 2001-2023 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.0
+ */
+
+export function clickGuard(element: HTMLElement, eventHandler: (event: MouseEvent) => Promise<void>): void {
+  let pending = false;
+
+  element.addEventListener("click", (event) => {
+    if (pending) {
+      event.preventDefault();
+
+      return;
+    }
```

Comment:
I'm not sure if this function should also bind the event listener. It might be useful to make this more generically reusable:

```
function promiseMutex<T extends ((...args: any[]) => Promise<unknown>)>(promise: T): ((...args: Parameters<T>) => boolean)
{
    let pending = false;

    return function (...args: Parameters<T>): boolean {
        if (pending) {
            return false;
        }
        pending = true;
        promise(...args).finally(() => {
            pending = false;
        });

        return true;
    };
}
```

and then:

```
button.addEventListener("click", promiseMutex(() => showDialog(button.dataset.url!)));
// or
const callback = promiseMutex(() => showDialog(button.dataset.url!));
button.addEventListener("click", (event) => {
  if (!callback(event)) {
     // already running
     event.preventDefault();
  }
});
```

see:

https://www.typescriptlang.org/play?target=8&ssl=16&ssc=2&pln=1&pc=1#code/GYVwdgxgLglg9mABABwE5wLYwM4FMCyIUuAHgDwAqipxYAJtogBRMB07AhqgObYBciDmACeAbQC6ASkQBeAHyIACuix4y4ANZg4AdzBzJcpmkw5cAipIEt2rLrwGKuHDLmKpslA7IUAjOHAANrhCkgBQAN5hiDGIwVAouPQwYNyyiMAcgXgA3GHRsahuIKhIoJCwCMy29vxKzq7unhQGAv5BIUhRsT2IMMDMyEl0KdzS3b2TRVAlZVm5BZMAvou9Q8mp6VCoILh5kzEmqrhsnDzYkqzAKVmBwizS8ogTB7HrI5syGfN7qz1Lkjyf0KxVKiG2u32sSWeRWYQgCGwCQAwgAhdJHMyEYgkJgcbDCSAZcDQeBIJgANyyuwEYBAGF8uFQViUKjMZApcBgdAUL0QCLASJQ6TAuB0rNMeBYRWwQQpuAANIgigArXDQR684ExPBQCgwVxwIgPHzK3CywLyynU3CSJUARgADM7AasAUCegKLbhWIE4Nw8ToODAEshXe78l6Or7-Uw0Ux7ZJXVHgjGA-GAExJvK6-WG41MTXPfmI6N+9OopgAZmziCWDoArC68kA